src/Processors/Formats/Impl/CSVRowInputFormat.cpp

```diff
@@ -402,6 +402,11 @@ bool CSVFormatReader::readFieldOrDefault(DB::IColumn & column, const DB::DataTyp
     String field;
     readCSVField(field, *buf, format_settings.csv);
     ReadBufferFromString tmp_buf(field);
+    if (tmp_buf.eof())
```

Comment:
But if we have `tmp_buf.eof()` it means that the field is empty, and if setting `input_format_csv_empty_as_default` is enabled (as it is by default), we should check it and insert default here:
https://github.com/ClickHouse/ClickHouse/blob/ddb33d6e35097729df0d75fe088ab463a45ae396/src/Processors/Formats/Impl/CSVRowInputFormat.cpp#L367-L382

Why it doesn't work in your case?

Reply:
As we have set `input_format_csv_empty_as_default`  as `false`, so we encounter this problem. And set it as true, has slove  our problems, thanks. I will close this pr.